feat: add Helmet middleware to secure http headers

emmadal opened the pull request:
Added a Helmet middleware to enhance security by setting important HTTP headers, protecting against common web vulnerabilities such as XSS, Clickjacking, MIME sniffing, and insecure connections.

**Content Security Policy (CSP)** : Restricts resource loading to prevent XSS attacks.
**X-Frame-Options**: Protects against Clickjacking by preventing iframe embedding.
**X-Content-Type-Options** : Prevents MIME-type sniffing attacks.
**X-XSS-Protection** : Enables browser’s built-in XSS filter.
**Strict Transport Security (HSTS)** : Enforces HTTPS for secure communication.
**Referrer-Policy** : Controls how much referrer info is shared.
**Cache-Control** : Prevents sensitive data from being cached.
**X-Powered-By** : Option to remove or customize to prevent exposure of server details.

jeffotoni commented:
@emmadal  
Thank you for your collaboration, we will check it out soon.
Thank you. 🚀🐧🥰

emmadal commented:
> @emmadal Thank you for your collaboration, we will check it out soon. Thank you. 🚀🐧🥰

Thanks @jeffotoni  it's a pleasure for me to contribute 🚀

jeffotoni commented:
@emmadal 
I will add and complete what started is fine for you ?

jeffotoni commented:
@emmadal 
I couldn't help it, I did it here.
First of all, thank you very much for your collaboration, it motivates us even more.
Sorry, but I had to change the names of some fields in the options and add them.
When we make any change or create middleware, for example, we research how fiber, echo and gin did it, and fiber is one of our inspirations, so I made it similar to theirs and made a comparison chart of what we had and what was missing and added them, so that we would be the same or better (laughs) whenever possible, of course.
Another point, now we can build native Quick middleware, so I converted it to quick.
And I saw your test, congratulations, you used q.Qtest, very good, in every quick project the goal is to have 0% zero external dependencies, I'm just reinforcing that.
Sorry, we haven't written a doc on how to contribute yet, all of this should be there, but there's so much that I can't be 100% complete, unfortunately. But we are very grateful for your collaboration and it allowed us to study and delve even deeper when we saw its implementation.

Thank you. 😍🐧🚀

jeffotoni commented:
![image](https://github.com/user-attachments/assets/061788eb-bb60-4f7c-b536-bab17e937c53)

jeffotoni commented:
@emmadal 
![image](https://github.com/user-attachments/assets/3f61b3ca-4098-47e2-949f-e9f4b57204b8)

jeffotoni commented:
It's not a good practice but I ended up correcting some points that I noticed when testing the helmet, don't do this at home (lol) 

![image](https://github.com/user-attachments/assets/8a9c2e6b-e667-42d4-bdb6-01cc0b4d61ae)

emmadal commented:
> @emmadal 
> I couldn't help it, I did it here.
> First of all, thank you very much for your collaboration, it motivates us even more.
> Sorry, but I had to change the names of some fields in the options and add them.
> When we make any change or create middleware, for example, we research how fiber, echo and gin did it, and fiber is one of our inspirations, so I made it similar to theirs and made a comparison chart of what we had and what was missing and added them, so that we would be the same or better (laughs) whenever possible, of course.
> Another point, now we can build native Quick middleware, so I converted it to quick.
> And I saw your test, congratulations, you used q.Qtest, very good, in every quick project the goal is to have 0% zero external dependencies, I'm just reinforcing that.
> Sorry, we haven't written a doc on how to contribute yet, all of this should be there, but there's so much that I can't be 100% complete, unfortunately. But we are very grateful for your collaboration and it allowed us to study and delve even deeper when we saw its implementation.
> 
> Thank you. 😍🐧🚀


Yeah i see. I  the future i will take this in consideration

emmadal commented:
@jeffotoni Thanks for you refactorisation and help. i think in the refactorisation. there are 2 fields which missing, like `X-Powered-By` to see which gives a details about the details of framework and `Server` to hide on which server the service is hosted  😎
What do you think ?

emmadal commented:
> @emmadal
> 
> Good point, you can send a pull request and I'll upload it.
> 
> With behavior like this: • If HidePoweredBy = true → remove X-Powered-By and Server • If PoweredBy ≠ “” → sets the value of X-Powered-By • If ServerHeader ≠ “” → sets the value of Server
> 
> Yes, for the next one, you can follow what was done in the helmet there it was in the quick pattern. Pattern in the comments Pattern in the tests Pattern using quick.Handler
> 
> If you need to implement http, you can implement http. and let us know that we will make the abstractions for quick.go or ctx.go ok? Because I didn't encapsulate all the possibilities of net/http we are still doing it as needed.

@jeffotoni  Ok no worries i will send a PR to update it and try to build the next middleware. also about the website does a way to contribute too ?
